src/plugins/dnf/subscription_manager.py

```diff
@@ -270,22 +270,26 @@ def _upload_profile(self) -> None:
         various reasons, then we try to do it ourselves in blocking way.
         """
         # First try to get PID of rhsmcertd from lock file
+        rhsmcertd_pid = None
         try:
             with open("/var/lock/subsys/rhsmcertd", "r") as lock_file:
                 rhsmcertd_pid = int(lock_file.readline())
         except (IOError, ValueError) as err:
             log.info(f"Unable to read rhsmcertd lock file: {err}")
```

Comment:
I would just add `self._upload_profile_blocking()` here. It should fix the issue.